### ISSUE TITLE
How to navigate selection item in action?

### ISSUE THREAD

**Issue by towry** [ibhagwan/fzf-lua#986]
> > @ibhagwan How to use `fzf_lua.fzf_exec` inside `setup.buffers.actions`? Or this is not recommended?
> 
> No problem, you can call whatever you want from the actions, it will override the existing fzf-lua interface, just make sure to wrap the function inside a table to prevent the “flash” as it tells fzf-lua to reuse the interface rather than close and reopen.

I am trying to bind ['ctrl-m'] = { function() end } in files actions, but when i presss 'ctrl-m', the fzf-window shows process exit 0. (window is not closed).

i am trying to integrate flash.nvim.

_Originally posted by @towry in https://github.com/ibhagwan/fzf-lua/issues/499#issuecomment-1879977737_
            
---- 
more context:

I use `{}` because i'd like fzf window do not close and use [flash.nvim](https://github.com/folke/flash.nvim?tab=readme-ov-file#-examples) to quickly jump to a selection.
So i don't have to use <c-n> or enter more keys to narrow down the filter to select.

**Comment by towry** [comment]
My currently almost work attempt is:

1. create `after/ftplugin/fzf.lua` and bind buffer local to mode 't' with key `<c-a>`.
2. in the keymap callback run:

```lua
  require('flash').jump({
    pattern = '^',
    label = { after = { 0, 0 } },
    search = {
      mode = 'search',
      exclude = {
        function(win)
          return false
        end,
      },
    },
    action = function(match)
      vim.print(match)
      -- local picker = require('telescope.actions.state').get_current_picker(prompt_bufnr)
      -- picker:set_selection(match.pos[1] - 1)
    end,
  })
```

The result is:

<img width="842" alt="截屏2024-01-07 20 58 44" src="https://github.com/ibhagwan/fzf-lua/assets/8279858/555cb8b6-696e-47ff-b9c1-4e71b36842b0">

So, i need to:

1. find the list window id, exclude other windows in `exclude` filter fn.
2. in action, move cursor line to the match.

**Comment by ibhagwan** [comment]
I’m not sure if I understand correctly what you’re trying to do, are you trying to have an action navigate within the fzf-lua selection?

If so, wrapping the action with `{}` won’t help as this just prevents from the float windows being closed but the fzf process is already dead (and its selections too), if you want the action to be executed while the fzf process is still running you need to use `{ fn = function() … end, exec_silent = true }` (this gets converted to fzf’s `execute-silent` action), similar to how the “copy commit hash” function is defined:
https://github.com/ibhagwan/fzf-lua/blob/5890b84b07f842da292ee9067b7363ac2af02d92/lua/fzf-lua/defaults.lua#L276

**Comment by towry** [comment]
Sorry for the my bad english. The original question is addressed.

The current question is, how to navigate(<down>, <up>) selection item in action.

**Comment by ibhagwan** [comment]
> In the above picture, if i press j, I want to move cursor to the line highlighted with char j, in the assumption that I already know the line number and col number of line j.

You can’t use `nvim_win_set_cursor` with `buffer=0` (current buffer) as I’m not sure which buffer context the action is called under, iterate neovim windows, find which one is a terminal buffer and use that buffer number with `nvim_win_set_cursor`.

**Comment by towry** [comment]
> use that buffer number with nvim_win_set_cursor.

But `nvim_win_set_cursor` use window id, not buffer id, I am a little confused.

**Comment by ibhagwan** [comment]
> > use that buffer number with nvim_win_set_cursor.
> 
> But `nvim_win_set_cursor` use window id, not buffer id, I am a little confused.

You’re correct, find the correct win id then :)

**Comment by towry** [comment]
I already have the correct win id.

**Comment by ibhagwan** [comment]
You can use `require'fzf-lua'.utils.fzf_winobj().fzf_winid` to get FzfWin’s class win id.

**Comment by towry** [comment]
> I already have the correct win id.

**Comment by ibhagwan** [comment]
> > I already have the correct win id.

Not sure what this quote means, did my last comment solve your issue?

**Comment by towry** [comment]
No, because I already have correct win id, and then you suggest me 

> You can use require'fzf-lua'.utils.fzf_winobj().fzf_winid to get FzfWin’s class win id.

But in my understanding it seems not possible because fzf main window is terminal window that runs fzf inside it, the `nvim_win_set_cursor` api can not change the fzf selection item. 

And I don't want to occupy too much of your time so closing this issue for now until i have more clues.

**Comment by Zevan770** [comment]
An unbelievable fact: fzf has it's builtin Easymotion-like function -- implemented far back in 2016, [version 0.12.2](https://github.com/junegunn/fzf/blob/master/CHANGELOG.md#0122) ! 

fzf --bind "alt-s:jump"

![image](https://cloud.githubusercontent.com/assets/700826/15367574/b3999dc4-1d64-11e6-85da-28ceeb1a9bc2.png)
